src/foam/dao/PgConnectionPool.java

```diff
@@ -0,0 +1,32 @@
+package foam.dao;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public abstract class PgConnectionPool {
```

Comment:
Yes, It's a generic JDBC database pool implementation, and yes it will work with any other relational database compatible with jdbc.

@kgrgreer , do you think the pool size number should be defined at runtime based on runtime properties, such as memory size , cpus etc ? also, do you think it should be something configurable on the model ?

https://commons.apache.org/proper/commons-dbcp/